### PR TITLE
feat(java): add version latestVersion api

### PR DESCRIPTION
**Pull request by LuQQiu**

### CODE REVIEW

File: java/core/src/test/java/com/lancedb/lance/TestUtils.java [hunk 1]
```diff
@@ -127,14 +129,18 @@ public void createDatasetAndValidate() throws IOException, URISyntaxException {
           assertEquals(ROW_COUNT, dataset.countRows());
           schema = reader.getVectorSchemaRoot().getSchema();
           validateFragments(dataset);
+          assertEquals(1, dataset.version());
+          assertEquals(1, dataset.latestVersion());
```
Comment (eddyxu) [inline java/core/src/test/java/com/lancedb/lance/TestUtils.java:133]: Do we have tests where `version() != latestVersion()`?

Comment (LuQQiu) [reply java/core/src/test/java/com/lancedb/lance/TestUtils.java:140]: Addressed the comment, PTAL

**Comment by LuQQiu** [comment]
@eddyxu PTAL, thanks!

**Comment by LuQQiu** [comment]
@eddyxu PTAL

File: java/core/lance-jni/src/error.rs [hunk 2]
```diff
@@ -41,8 +41,8 @@ impl JavaException {
 pub enum Error {
     #[snafu(display("JNI error: {}", message))]
     Jni { message: String },
-    #[snafu(display("Invalid argument: {}", message))]
-    InvalidArgument { message: String },
+    #[snafu(display("Invalid argument: {}, location: {}", message, location))]
+    InvalidArgument { message: String, location: Location },
```
Comment (QianZhu) [inline java/core/lance-jni/src/error.rs:45]: why do we need Location in the error msg?

**Review by QianZhu — APPROVED** [review]

File: java/core/lance-jni/src/error.rs (hunk 2 above)
Comment (chebbyChefNEQ) [reply java/core/lance-jni/src/error.rs:45]: this location tells us which line generated the error. This is how we get a "stack track" when an error happens

File: java/core/lance-jni/src/error.rs [hunk 3]
```diff
@@ -97,6 +97,7 @@ impl From<Utf8Error> for Error {
     fn from(source: Utf8Error) -> Self {
         Self::InvalidArgument {
             message: source.to_string(),
+            location: location!(),
```
Comment (chebbyChefNEQ) [inline java/core/lance-jni/src/error.rs:100]: This should use `track_caller` location instead of the conversion site location.

Comment (chebbyChefNEQ) [reply java/core/lance-jni/src/error.rs:100]: see example here: https://github.com/lancedb/lance/blob/main/rust/lance-core/src/error.rs#L154

Comment (LuQQiu) [reply java/core/lance-jni/src/error.rs]: Thanks for the example @chebbyChefNEQ ! Addressed the comments, PTAL

**Review by chebbyChefNEQ — APPROVED** [review]

**Pull request merged**